gym_multigrid/envs/lavarooms.py

```diff
@@ -0,0 +1,332 @@
+
+
+Observation: TypeAlias = ObservationDict | MultiAgentObservationDict | NDArray[np.int_]
+
+
+class LavaRooms(MultiGridEnv):
+    """
+    Environment for capture the flag with multiple agents with N blue agents and M red agents.
+    """
+
+    def __init__(
+        self,
+        env_seed: int = 0,
+        grid_size: tuple = (9, 9),
+        agent_view_size: int = 7,
+        max_steps: int = 100,
+        tile_size: int = 20,
+        highlight_visible_cells: bool | None = True,
+        partial_observability: bool = False,
+        render_mode: Literal["human", "rgb_array"] = "rgb_array",
+    ) -> None:
+        """
+        Initialize a new capture the flag environment.
+
+        Parameters
+        ----------
+
+        """
+        if env_seed < 0 or env_seed >= 2:
+            raise ValueError(
+                f"The Lavaroom only accepts env_seed of 0 and 1, given {env_seed}"
+            )
+        else:
+            self.env_seed = env_seed
+
+        self.width = grid_size[0]
+        self.height = grid_size[1]
+        self.grid_size = grid_size
+        self.max_steps = max_steps
+        self.world = FRWorld
+        self.actions_set = FRActions
+
+        see_through_walls: bool = False
+
+        self.agents = [
+            Agent(
+                self.world,
+                color="blue",
+                bg_color="light_blue",
+                view_size=agent_view_size,
+                actions=self.actions_set,
+                type="agent",
+            )
+        ]
+
+        self.doorway_positions = [(10, 6), (2, 6)]
+        self.vert_wall_positions = [(6, 0), (7, 6)]
+        self.hor_wall_positions = [(0, 6), (6, 6)]
+
+        self.goal_positions = [(6, 3), (1, 1)]  # (7, 1)
+        self.agent_positions = [(6, 9), (1, 11)]
+        self.lava_positions = [
+            [
+                (2, 6),
+                (2, 7),
+                (2, 8),
+                (3, 7),
+                (3, 8),
+                (4, 7),
+                (4, 8),
+            ],
+            [
+                (10, 6),
+                (10, 7),
+                (10, 8),
+                (9, 7),
+                (9, 8),
+                (8, 7),
+                (8, 8),
+            ],
+        ]
+
+        super().__init__(
+            width=self.width,
+            height=self.height,
+            max_steps=max_steps,
+            see_through_walls=see_through_walls,
+            agents=self.agents,
+            agent_view_size=agent_view_size,
+            actions_set=self.actions_set,
+            partial_obs=partial_observability,
+            world=self.world,
+            render_mode=render_mode,
+            highlight_visible_cells=highlight_visible_cells,
+            tile_size=tile_size,
+        )
+
+    def _gen_grid(self, width, height):
+        # Create the grid
+        self.grid = Grid(width, height, self.world)
+
+        # Generate the surrounding walls
+        self.grid.horz_wall(0, 0)
```

Comment:
for the readability, use self.grid.rect_wall() method to generate the surrounding walls